src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs

```diff
@@ -59,8 +59,47 @@ private void Start()
         {
             _logger.LogInformation($"Starting {nameof(LinuxContainerActivityPublisher)}");
 
-            // start the timer by setting the due time
-            SetTimerInterval(_initialFlushIntervalMs);
+            _publishingTask = StartPublishingAsync();
+        }
+
+        private async Task StartPublishingAsync()
+        {
+            _publishingCts = new CancellationTokenSource();
```

Comment:
Since this is only set in one place, should it be done ctor and made `readonly`?

Reply:
Debated on that. Though this gives us the ability to defer initialization to when needed only, in practice, that will almost never be the case.

Reply:
I'd feel better making that change if we actually prevented multiple start/stop calls in this type, though.